File: src/dataset/c/armstrong-numbers_exercism-viettronics868/gemini/GeminiFlash_armstrong_numbers_v3.c
#include "armstrong_numbers.h"
#include <stdint.h>
#include <stdbool.h>

bool is_armstrong_number(int candidate) {
    if (candidate < 0) {
        return false;
    }

    int num_digits = 0;
    int temp = candidate;
    while (temp > 0) {
        temp /= 10;
        num_digits++;
    }

    int sum = 0;
    temp = candidate;
    while (temp > 0) {
        int digit = temp % 10;
        int power = 1;
        for (int i = 0; i < num_digits; i++) {
            power *= digit;
        }
        if (sum > INT_MAX - power) {
            return false;
        }
        sum += power;
        temp /= 10;
    }

    return sum == candidate;
}

File: src/dataset/javascript/armstrong-numbers_exercism-SLIB53/gemini/GeminiFlash_armstrong-numbers_v3.js
const numDigits = integer => {
  return Math.floor(Math.log10(integer)) + 1;
};

const validate = integer => {
  const num_digits = numDigits(integer);
  let sum = 0;
  let temp = integer;

  for (let i = num_digits - 1; i >= 0; i--) {
    const digit = Math.floor(temp / (10 ** i));
    sum += digit ** num_digits;
    temp %= (10 ** i);
  }

  return integer === sum;
};

export { validate };